### PR TITLE
initial kernel config changes for criu

### PR DESCRIPTION
**Pull request by NikolaBo**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR enables kernel configurations required for [CRIU ](https://criu.org/Main_Page) to run on mariner.

###### Change Log  

- Turned on `CONFIG_CHECKPOINT_RESTORE`, `CONFIG_BINFMT_MISC`, `CONFIG_MEM_SOFT_DIRTY`, `CONFIG_PACKET_DIAG`, `CONFIG_UNIX_DIAG`, `CONFIG_INET_DIAG`, `CONFIG_INET_TCP_DIAG`, `CONFIG_INET_UDP_DIAG`, `CONFIG_NETFILTER_XT_MARK`, `CONFIG_STP`, `CONFIG_BRIDGE`, `CONFIG_LLC`, `CONFIG_NETLINK_DIAG`, `CONFIG_MACVLAN` and `CONFIG_TUN`.
- Set `CONFIG_INET_SCTP_DIAG` to `m`.
- Added `CONFIG_INET_RAW_DIAG` and `CONFIG_INET_DIAG_DESTROY` not set.

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

Locally built, installed on vm, and tested.

### CODE REVIEW

**Comment by Camelron** [comment]
Mirrored code from fork and ran pipeline to test: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=233401&view=results

File: SPECS/kernel/config [hunk 1]
```diff
@@ -208,7 +208,7 @@ CONFIG_IPC_NS=y
 CONFIG_USER_NS=y
 CONFIG_PID_NS=y
 CONFIG_NET_NS=y
-# CONFIG_CHECKPOINT_RESTORE is not set
+CONFIG_CHECKPOINT_RESTORE=y
```
Comment (rlmenge) [inline SPECS/kernel/config:211]: Out of curiosity, do you know if this increases the size of the rpm significantly?

**Review by neha170 — APPROVED** [review]

**Review by rlmenge — APPROVED** [review]

File: SPECS/kernel/config (hunk 1 above)
Comment (NikolaBo) [reply SPECS/kernel/config:211]: RPM size is about 35M before and after the change.
Before: 36276851
After: 36312290

Comment (NikolaBo) [reply SPECS/kernel/config:211]: Kernel boot time is also essentially unchanged: `Startup finished in 1.364s (kernel)` with the changes and `Startup finished in 1.362s (kernel)` without (from `systemd-analyze`).

**Pull request merged**